src/components/Tooltip/Tooltip.tsx

```diff
@@ -56,113 +56,221 @@ const paddingsBottom = {
+    const setArrowClasses = () => {
+        const currentPlacement = popperInstance.state?.placement ?? position;
+        switch (true) {
+            case currentPlacement.toString().includes(TooltipPosition.Top):
+                return "before:tw-border-t-0 before:tw-border-l-0 tw-bottom-[-6px] before:tw-bg-black-100 before:dark:tw-bg-white";
+            case currentPlacement.toString().includes(TooltipPosition.Right):
+                return merge([
+                    "before:tw-border-t-0 before:tw-border-r-0 tw-left-[-6px]",
+                    brightHeader && alignment === TooltipAlignment.Start
+                        ? brightHeaderArrowBackgroundColors[brightHeader]
+                        : "before:tw-bg-black-100 before:dark:tw-bg-white",
+                ]);
+            case currentPlacement.toString().includes(TooltipPosition.Bottom):
+                return merge([
+                    "before:tw-border-b-0 before:tw-border-r-0 tw-top-[-6px]",
+                    brightHeader
+                        ? brightHeaderArrowBackgroundColors[brightHeader]
+                        : "before:tw-bg-black-100 before:dark:tw-bg-white",
+                ]);
+            case currentPlacement.toString().includes(TooltipPosition.Left):
+                return merge([
+                    "before:tw-border-b-0 before:tw-border-l-0 tw-right-[-6px]",
+                    brightHeader && alignment === TooltipAlignment.Start
+                        ? brightHeaderArrowBackgroundColors[brightHeader]
+                        : "before:tw-bg-black-100 before:dark:tw-bg-white",
+                ]);
+            default:
+                return "before:tw-border-b-0 before:tw-border-r-0 tw-top-[-6px] before:tw-bg-black-100 before:dark:tw-bg-white";
+        }
+    };
+
+    const tooltipOffset = withArrow ? 10 : 5;
+    const popperInstance = usePopper(triggerRefElement?.current, tooltipContainerRef.current, {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        placement: placement,
+        modifiers: [
+            {
+                name: "arrow",
+                options: {
+                    element: arrowElement,
+                },
+            },
+            {
+                name: "offset",
+                options: {
+                    offset: [0, tooltipOffset],
+                },
+            },
+            {
+                name: "flip",
+                enabled: flip,
+            },
+        ],
+    });
+
+    const arrowStyling = setArrowClasses();
+    const [isOpen, setIsOpen] = useState(false);
+
+    useEffect(() => {
+        if (triggerRefElement) {
+            triggerRefElement.current.addEventListener("click", () => setIsOpen(!isOpen));
+            triggerRefElement.current.addEventListener("mouseover", () => setIsOpen(true));
+            triggerRefElement.current.addEventListener("mouseleave", () => setTimeout(() => setIsOpen(false), 2000));
```

Comment:
We either need to clear the timeout if the mouse is on the tooltip, either add a new variable e.g. `mouseOverTooltip` that would be true while the user is over the tooltip and which conditions the display of the tooltip as well. So the condition to display the tooltip would be then: `isOpen && mouseOverTooltip`. Otherwise the tooltip closes even if the user is on over the tooltip with the mouse.